share/normalize/links.py

```diff
@@ -534,7 +536,89 @@ def execute(self, obj):
         raise Exception('All chains failed {}'.format(errors))
 
 
-class OrcidLink(AbstractLink):
+class AbstractIRILink(AbstractLink):
+    """Normalize IRIs
+
+    """
+    RULES = 'absolute_IRI'
+    SAFE_SEGMENT_CHARS = ":@-._~!$&'()*+,;="  # https://github.com/gruns/furl/blob/master/furl/furl.py#L385
+
+    @classmethod
+    def hint(cls, obj):
+        """A percentage expressed as a float of how likely a the given object can be parsed as this class
+        """
+        raise NotImplementedError
+
+    def execute(self, obj):
+        if not isinstance(obj, str):
+            raise TypeError('\'{}\' is not of type str.'.format(obj))
+
+        parsed = self._parse(obj)
+        parsed = self._process(**parsed)
+        return rfc3987.parse(rfc3987.compose(**parsed))
```

Comment:
In case we want to decompose it somewhere? But we do plan on just parsing them out rather than having them sent to us...